cg/meta/deliver/delivery_api.py

```diff
@@ -0,0 +1,245 @@
+"""Module for deliveries of workflow files"""
+
+import logging
+from pathlib import Path
+from typing import Iterable
+
+from housekeeper.store.models import File, Version
+
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.constants import delivery as constants
+from cg.constants.constants import DataDelivery
+from cg.exc import MissingFilesError
+from cg.meta.deliver.utils import (
+    create_link,
+    get_delivery_case_name,
+    get_delivery_dir_path,
+    get_case_tags_for_pipeline,
+    get_out_path,
+    get_sample_out_file_name,
+    get_sample_tags_for_pipeline,
+    include_file_case,
+    include_file_sample,
+)
+from cg.store import Store
+from cg.store.models import Case, CaseSample, Sample
+
+LOG = logging.getLogger(__name__)
+
+
+class DeliveryAPI:
+    """Deliver API for workflows files."""
+
+    def __init__(
+        self,
+        store: Store,
+        hk_api: HousekeeperAPI,
+        project_base_path: Path,
+        force_all: bool = False,
+        ignore_missing_bundles: bool = False,
+        dry_run: bool = False,
+    ):
+        self.store = store
+        self.hk_api = hk_api
+        self.project_base_path = project_base_path
+        self.ignore_missing_bundles = ignore_missing_bundles
+        self.deliver_failed_samples = force_all
+        self.dry_run = dry_run
+
+    def set_dry_run(self, dry_run: bool):
+        self.dry_run = dry_run
+
+    def set_deliver_failed_samples(self, deliver_failed_samples: bool):
+        self.deliver_failed_samples = deliver_failed_samples
+
+    def set_ignore_missing_bundles(self, ignore_missing_bundles: bool):
+        self.ignore_missing_bundles = ignore_missing_bundles
+
+    def deliver(self, ticket: str | None, case_id: str | None, pipeline: str) -> None:
+        if ticket:
+            self._deliver_files_by_ticket(ticket=ticket, pipeline=pipeline)
+        elif case_id:
+            self._deliver_files_by_case(case_id=case_id, pipeline=pipeline)
+
+    def _deliver_files_by_ticket(self, ticket: str, pipeline: str) -> None:
+        cases: list[Case] = self.store.get_cases_by_ticket_id(ticket)
+        if not cases:
+            LOG.warning(f"Could not find cases for ticket {ticket}")
+            return
+        for case in cases:
+            self.deliver_files(case=case, pipeline=pipeline)
+
+    def _deliver_files_by_case(self, case_id: str, pipeline: str) -> None:
+        case: Case | None = self.store.get_case_by_internal_id(case_id)
+        if not case:
+            LOG.warning(f"Could not find case {case_id}")
+            return
+        self.deliver_files(case=case, pipeline=pipeline)
+
+    def deliver_files(self, case: Case, pipeline: str):
+        if not self._case_is_deliverable(case=case, pipeline=pipeline):
+            return
+        self._deliver_case_files(case=case, pipeline=pipeline)
+        self._deliver_sample_files(case=case, pipeline=pipeline)
```

Comment:
```suggestion
        if self._case_is_deliverable(case=case, pipeline=pipeline):
            self._deliver_case_files(case=case, pipeline=pipeline)
            self._deliver_sample_files(case=case, pipeline=pipeline)
```

Reply:
I prefer returning early to not introduce unnecessary indentation

Reply:
I think it is cleaner to indent it like this since it is only a few lines 👍